### PR TITLE
Add macros for Set.every and Map.every

### PR DESCRIPTION
**Pull request by Dionysusnu**
Discussed on discord: https://discordapp.com/channels/476080952636997633/494540040991539200/707927393125269605
Transformed Array.every into function to reduce code repetition.

### CODE REVIEW

**Comment by Dionysusnu** [comment]
I'm not very good at git... Is there a way to not include that "Merge from roblox-ts/refactor"? Or is there a way to not generate that commit at all?

**Comment by Dionysusnu** [comment]
Oh, this will also fail to compile projects because @rbxts/types does not include these methods yet

File: src/TSTransformer/macros/propertyCallMacros.ts [hunk 1]
```diff
@@ -90,55 +90,49 @@ const STRING_CALLBACKS: MacroList<PropertyCallMacro> = {
...
+function makeEveryOrSomeMethod(
+	iterator: lua.Identifier,
+	callbackArgsListMaker: (
+		keyId: lua.TemporaryIdentifier,
+		valueId: lua.TemporaryIdentifier,
+		expression: lua.Expression,
+	) => lua.List<lua.Expression>,
+	hasInvertedCallback: boolean,
+	startBool: boolean,
+	endBool: boolean,
```
Comment (osyrisrblx) [inline src/TSTransformer/macros/propertyCallMacros.ts:102]: endBool is always !startBool, do we need this argument?

File: src/TSTransformer/macros/propertyCallMacros.ts [hunk 2]
```diff
@@ -90,55 +90,49 @@ const STRING_CALLBACKS: MacroList<PropertyCallMacro> = {
...
 		expression = state.pushToVarIfComplex(expression);
 
-		const resultId = state.pushToVar(lua.bool(true));
+		const resultId = state.pushToVar(lua.bool(startBool));
 		const callbackId = state.pushToVarIfComplex(transformExpression(state, node.arguments[0]));
 
-		const keyId = lua.tempId();
-		const valueId = lua.tempId();
+		const firstId = lua.tempId();
+		const secondId = lua.tempId();
```
Comment (osyrisrblx) [inline src/TSTransformer/macros/propertyCallMacros.ts:111]: I think keyId, valueId were more descriptive names for these. I can't think of a scenario where we're using something other than pairs/ipairs.

File: src/TSTransformer/macros/propertyCallMacros.ts [hunk 3]
```diff
@@ -90,55 +90,49 @@ const STRING_CALLBACKS: MacroList<PropertyCallMacro> = {
...
-
-	every: (state, node, expression) => {
+function makeEveryOrSomeMethod(
+	iterator: lua.Identifier,
+	callbackArgsListMaker: (
+		keyId: lua.TemporaryIdentifier,
+		valueId: lua.TemporaryIdentifier,
+		expression: lua.Expression,
+	) => lua.List<lua.Expression>,
+	hasInvertedCallback: boolean,
```
Comment (osyrisrblx) [inline src/TSTransformer/macros/propertyCallMacros.ts:100]: We could base this off of startBool?

File: src/TSTransformer/macros/propertyCallMacros.ts [hunk 4]
```diff
@@ -148,42 +142,59 @@ const READONLY_ARRAY_METHODS: MacroList<PropertyCallMacro> = {
...
+		const separator = node.arguments.length > 0 ? transformExpression(state, node.arguments[0]) : lua.string(", ");
+		return lua.create(lua.SyntaxKind.CallExpression, {
+			expression: lua.globals.table.concat,
+			args: lua.list.make(expression, separator),
+		});
 	},
 
+	every: makeEveryMethod(lua.globals.ipairs, (keyId, valueId, expression) => {
+		return lua.list.make(valueId, offset(keyId, -1), expression);
+	}),
```
Comment (osyrisrblx) [inline src/TSTransformer/macros/propertyCallMacros.ts:192]:
```suggestion
	every: makeEveryMethod(lua.globals.ipairs, (keyId, valueId, expression) => lua.list.make(valueId, offset(keyId, -1), expression)),
```

File: src/TSTransformer/macros/propertyCallMacros.ts [hunk 5]
```diff
@@ -415,8 +426,32 @@ const ARRAY_METHODS: MacroList<PropertyCallMacro> = {
 	},
 };
 
+const READONLY_SET_METHODS: MacroList<PropertyCallMacro> = {
+	/*
+	every: makeEveryMethod(lua.globals.pairs, (firstId, secondId, expression) => {
+		return lua.list.make(firstId, expression);
+	}),
```
Comment (osyrisrblx) [inline src/TSTransformer/macros/propertyCallMacros.ts:433]:
```suggestion
	every: makeEveryMethod(lua.globals.pairs, (firstId, secondId, expression) => lua.list.make(firstId, expression)),
```

File: src/TSTransformer/macros/propertyCallMacros.ts [hunk 6]
```diff
@@ -415,8 +426,32 @@ const ARRAY_METHODS: MacroList<PropertyCallMacro> = {
...
+
+	some: makeSomeMethod(lua.globals.pairs, (firstId, secondId, expression) => {
+		return lua.list.make(firstId, expression);
+	}),
```
Comment (osyrisrblx) [inline src/TSTransformer/macros/propertyCallMacros.ts:437]:
```suggestion
	some: makeSomeMethod(lua.globals.pairs, (firstId, secondId, expression) => lua.list.make(firstId, expression)),
```

File: src/TSTransformer/macros/propertyCallMacros.ts [hunk 7]
```diff
@@ -415,8 +426,32 @@ const ARRAY_METHODS: MacroList<PropertyCallMacro> = {
...
+	*/
+};
+
 const SET_METHODS: MacroList<PropertyCallMacro> = {};
 
+const READONLY_MAP_METHODS: MacroList<PropertyCallMacro> = {
+	/*
+	every: makeEveryMethod(lua.globals.pairs, (keyId, valueId, expression) => {
+		return lua.list.make(valueId, keyId, expression);
+	}),
```
Comment (osyrisrblx) [inline src/TSTransformer/macros/propertyCallMacros.ts:447]:
```suggestion
	every: makeEveryMethod(lua.globals.pairs, (keyId, valueId, expression) => lua.list.make(valueId, keyId, expression)),
```

File: src/TSTransformer/macros/propertyCallMacros.ts [hunk 8]
```diff
@@ -415,8 +426,32 @@ const ARRAY_METHODS: MacroList<PropertyCallMacro> = {
...
+
+	some: makeSomeMethod(lua.globals.pairs, (keyId, valueId, expression) => {
+		return lua.list.make(valueId, keyId, expression);
+	}),
```
Comment (osyrisrblx) [inline src/TSTransformer/macros/propertyCallMacros.ts:451]:
```suggestion
	some: makeSomeMethod(lua.globals.pairs, (keyId, valueId, expression) => lua.list.make(valueId, keyId, expression)),
```

File: src/TSTransformer/macros/propertyCallMacros.ts [hunk 9]
```diff
@@ -148,42 +142,59 @@ const READONLY_ARRAY_METHODS: MacroList<PropertyCallMacro> = {
...
+		});
 	},
 
+	every: makeEveryMethod(lua.globals.ipairs, (keyId, valueId, expression) => {
+		return lua.list.make(valueId, offset(keyId, -1), expression);
+	}),
+
+	some: makeSomeMethod(lua.globals.ipairs, (keyId, valueId, expression) => {
+		return lua.list.make(valueId, offset(keyId, -1), expression);
+	}),
```
Comment (osyrisrblx) [inline src/TSTransformer/macros/propertyCallMacros.ts:196]:
```suggestion
	some: makeSomeMethod(lua.globals.ipairs, (keyId, valueId, expression) => lua.list.make(valueId, offset(keyId, -1), expression)),
```

File: src/TSTransformer/macros/propertyCallMacros.ts (hunk 2 above)
Comment (Dionysusnu) [reply src/TSTransformer/macros/propertyCallMacros.ts:111]: Set uses it the opposite way around. ValueId, ContainsId. Initially wrote it for set and then converted it into this function, never changed the names back.

File: src/TSTransformer/macros/propertyCallMacros.ts (hunk 4 above)
Comment (Dionysusnu) [reply src/TSTransformer/macros/propertyCallMacros.ts:192]: Prettier doesn't like that...

**Pull request merged**